### PR TITLE
Added endpoint to get submitted file

### PR DESCRIPTION
**Pull request by BogVin**

### CODE REVIEW

File: app/submissions/views.py [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
+from wsgiref.util import FileWrapper
+from django.shortcuts import get_object_or_404
 
 
 class SubmissionViewList(viewsets.ModelViewSet):
     serializer_class = SubmissionSerializer
     queryset = Submission.objects.all()
+
+
+class FileDownloadListAPIView(generics.ListAPIView):
```
Comment (Chay23) [inline app/submissions/views.py:15]: Actually, I don't believe it won't l be able to handle the traffic load, that's expected.
I suggest to rewrite this part in Java...

File: app/submissions/urls.py [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
-from .views import SubmissionViewList
+from .views import SubmissionViewList, FileDownloadListAPIView
+from django.urls import path
 
 router = routers.DefaultRouter()
 router.register("", SubmissionViewList, basename="submissions")
-urlpatterns = router.urls
+urlpatterns = [
+    path(
+        "file/<str:folder>/<str:year>/<str:month>/<str:day>/<str:filename>/",
```
Comment (Chay23) [inline app/submissions/urls.py:9]: It's not acceptable.
Neither in java, not in python.

File: app/core/settings.py [hunk 3]
```diff
@@ -22,6 +22,7 @@
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 MEDIA_ROOT = os.path.join(BASE_DIR, "media")
+MEDIA_URL = "/media/"
```
Comment (Chay23) [inline app/core/settings.py:25]: I will not approve it, even if you buy me a beer...

File: app/submissions/views.py [hunk 4]
```diff
@@ -1,9 +1,32 @@
...
+    def get(self, request, *args, **kwargs):
+        folder = kwargs.get("folder", "")
+        year = kwargs.get("year", "")
+        month = kwargs.get("month", "")
```
Comment (Chay23) [inline app/submissions/views.py:19]: How long did you think about it?

File: app/assignments/migrations/0001_initial.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Generated by Django 3.1.2 on 2020-12-15 21:02
+# Generated by Django 3.1.2 on 2020-12-24 12:37
```
Comment (vladhoi) [inline app/assignments/migrations/0001_initial.py:1]: Please remove from pr all the migrations that don't affect actual tables

File: app/submissions/views.py [hunk 6]
```diff
@@ -1,9 +1,32 @@
...
+        day = kwargs.get("day", "")
+        filename = kwargs.get("filename", "")
+        attached_file = f"{folder}/{year}/{month}/{day}/{filename}"
+        queryset = get_object_or_404(Submission, attached_file=attached_file)
+        file_handle = queryset.attached_file.open()
+        document = file_handle
```
Comment (Chay23) [inline app/submissions/views.py:25]: Why did you create file_handle var?

File: app/tests/submissions/test_views.py [hunk 7]
```diff
@@ -66,3 +67,28 @@ def test_post_submission_failure(
...
+
+
+@pytest.mark.django_db
+def test_file_response(user_client, create_valid_submission):
+    """
+    Ensure we get  file.
+    """
+    create_valid_submission
+    submissions = Submission.objects.all()
+    submission = submissions[0]
```
Comment (Chay23) [inline app/tests/submissions/test_views.py:79]: No needs of this var

File: app/media/labs/2020/12/17/blob [hunk 8]
```diff
@@ -0,0 +1 @@
+# type your code...
```
Comment (vladhoi) [inline app/media/labs/2020/12/17/blob:1]: Please remove all media files from pr, we do not need them on github

**Review by vladhoi — CHANGES_REQUESTED** [review]

File: app/submissions/views.py (hunk 1 above)
Comment (BogVin) [reply app/submissions/views.py:15]: image

**Pull request closed without merging**